src/client/kotlin/tech/thatgravyboat/skyblockapi/impl/events/PacketEventHandler.kt

```diff
@@ -36,17 +36,20 @@ object PacketEventHandler {
                     ContainerInitializedEvent(event.packet.items, container.title).post()
                 }
             }
+
             is ClientboundContainerSetSlotPacket -> {
                 McClient.tell {
                     val containerId = event.packet.containerId
                     when (containerId) {
                         PLAYER_HOTBAR_CONTAINER_ID -> {
                             PlayerHotbarChangeEvent(event.packet.slot - FIRST_HOTBAR_SLOT, event.packet.item).post()
                         }
+
                         PLAYER_INVENTORY_CONTAINER_ID -> PlayerInventoryChangeEvent(event.packet.slot, event.packet.item).post()
                         else -> {
                             val container = McScreen.asMenu?.takeIf { it.menu?.containerId == containerId } ?: return@tell
-                            ContainerChangeEvent(event.packet.item, event.packet.slot, container.title).post()
+                            val allItems = container.menu?.slots?.map { it.item } ?: emptyList()
```

Comment:
the packet wont be proceeded here so `allItems` will contain the items before the change